### ISSUE TITLE
FIM Windows Agent - Case: Test Report Changes

### ISSUE THREAD

**Issue by damarisg** [wazuh/wazuh-qa#1886]
## Description 

FIM tests has a lot of tests to fix because fails inconsistent times.

| Related to | 
|--|
| https://github.com/wazuh/wazuh-qa/issues/1873

### Pipeline parameters

|  Jenkins branch  | QA branch   | 
|- |- |
|  v4.2.1-rc1 |   1873-4.2.1-fim-windows-agent |

## Packages Info

| Type | Format | Architecture | Versión | Revision |  Tag | File name |
|:--:|:--:|:--:|:--:|:--:|:--:|:--:|
| Server | rpm | x86_64 | v4.2.1 | 40214| v4.2.1 | wazuh-manager-4.2.1-0.1873.x86_64.rpm |
| Agent | rpm | x86_64 | v4.2.1 | 40214|v4.2.1| wazuh-agent-4.2.1-0.1873.x86_64.rpm |
| Agent | msi (Windows) | x86_64 | v4.2.1 | 40214| v4.2.1| wazuh-agent-4.2.1-0.1873.msi |
| Agent | pkg (macOS) | x86_64 | v4.2.1 | 40214| v4.2.1| wazuh-agent-4.2.1-0.1873.pkg |
|Agent | p5p (Solaris 11) | x86_64 | v4.2.1 | 40214|v4.2.1| wazuh-agent_v4.2.1-0.1873-sol11-i386.p5p​|
|Agent | p5p (Solaris 10) | x86_64 | v4.2.1 | 40214|v4.2.1| wazuh-agent_v4.2.1-0.1873-sol10-i386.pkg
|Agent | wpk (Linux) | x86_64 | v4.2.1 | 40214| v4.2.1 | wazuh_agent_v4.2.1-0.1873_linux_x86_64.wpk
|Agent | wpk (Windows) | x86_64 | v4.2.1 | 40214|v4.2.1| wazuh_agent_v4.2.1-0.1873_windows.wpk

**Note**: The tag is `v4.2.1` but for now, to create it on jenkins you need to use `v4.2.1-rc1`.

#### Manager
```
syscheck.debug=2
analysisd.debug=2
monitord.rotate_log=0
```

#### Agent
```
agent.debug=2
syscheck.debug=2
monitord.rotate_log=0
```

### Environment 

Provider| Box| OS| CPU| Memory | 
--|--|--|--|--|
Vagrant |  | Windows | 2 | 1024 |

### pytest_args
`-v --tier 0 --tier 1 --tier 2 --fim_mode="realtime" --fim_mode="whodata"`

## Development

You need to check that `test_fim/test_files/test_report_changes/` folder works successfully on Windows. Sometimes there are failures related to the tests :
- `test_fim\test_files\test_report_changes\test_report_deleted_diff.py: test_report_when_deleted_directories`. 
- `test_fim\test_files\test_report_changes\test_file_size_disabled.py`
- `test_fim\test_files\test_report_changes\test_large_changes.py`

If the tests work successfully after 2 times, you should see if other member could be verify that it woks successfully. 
But, if it fail you need to research and fix it.

## Tasks
- [x] Research about the Test Case.
- [x] Fix the Test Case.
- [x] Verify that each execution works successfully. (Full Green - Attach evidence.)
- [x] Verify that cover `scheduled`, `realtime`, and  `whodata`. Research if this case require all args, or only one.

**Comment by MizugorouZ** [comment]
### Local Results

The following are the results of the tests above with all the parameters for `fim_mode`. In each zip there are 7 HTML reports which match with each combination of `fim_mode`:
* `Results_TestName_rt` &rarr; `--fim_mode="realtime"`
* `Results_TestName_who` &rarr; `--fim_mode="whodata"`
* `Results_TestName_sch` &rarr; `--fim_mode="scheduled"`

| Test | Results | Date | By | Status |
|:--:|:--:|:--:|:--:|:--:|
| `test_fim/test_files/test_report_changes/test_report_deleted_diff.py` | [ResultsReportDeletedDiff](https://github.com/wazuh/wazuh-qa/files/7219624/ResultsReportDeletedDiff.zip) | 2021/09/23 | @MizugorouZ | :green_circle: |
| `test_fim/test_files/test_report_changes/test_file_size_disabled.py` | [ResultsFileSizeDisabled](https://github.com/wazuh/wazuh-qa/files/7219643/ResultsFileSizeDisabled.zip) | 2021/09/23 | @MizugorouZ | :green_circle: |
| `test_fim/test_files/test_report_changes/test_large_changes.py` | [ResultsLargeChanges](https://github.com/wazuh/wazuh-qa/files/7219663/ResultsLargeChanges.zip) | 2021/09/23 | @MizugorouZ | :green_circle: |

In addition, I also executed the whole suite `test_fim/test_files/test_report_changes/` to verify everything was OK:
| Test | Results | Date | By | Status |
|:--:|:--:|:--:|:--:|:--:|
| `test_fim/test_files/test_report_changes/` | [ResultsReportChanges](https://github.com/wazuh/wazuh-qa/files/7219681/ResultsReportChanges.zip) | 2021/09/23 | @MizugorouZ | :green_circle: |

**Comment by danisan90** [comment]
**Local Results**


These are the results of executing  `test_fim\test_files\test_report_changes` with the parameters `--fim_mode="realtime" --fim_mode="whodata" --fim_mode="scheduled"`
| Results |Date| By | Status |
|:--:|:--:|:--:|:--:|
| [ResultsReportChanges1](https://github.com/wazuh/wazuh-qa/files/7226940/ResultsReportChanges1.zip)| 2021/09/24 |@danisan90|  🔴 
|[ResultsReportChanges2](https://github.com/wazuh/wazuh-qa/files/7226951/ResultsReportChanges2.zip)| 2021/09/24 |@danisan90|  🔴 
| [ResultsReportChanges1a](https://github.com/wazuh/wazuh-qa/files/7226946/ResultsReportChanges1a.zip) | 2021/09/24 | @danisan90| 🔴
|[ResultsReportChanges2a](https://github.com/wazuh/wazuh-qa/files/7226957/ResultsReportChanges2a.zip)| 2021/09/24 | @danisan90| 🔴

**Comment by Deblintrake09** [comment]
**Local Results**

These are the results of executing test_fim\test_files\test_report_changes with the parameters --fim_mode="realtime" --fim_mode="whodata" --fim_mode="scheduled" 

| Results | Date | By | Status |Note|
|:--:|:--:|:--:|:--:|:--:|
| [ResultsReportChanges-PreFix1.zip](https://github.com/wazuh/wazuh-qa/files/7259163/ResultsReportChanges-PreFix1.zip) | 2021/09/29 | @Deblintrake09 | :red_circle: | without deactivating modules SCA, Rootcheck, Syscollector or Active Response.
| [ResultsReportChanges-PreFix.zip](https://github.com/wazuh/wazuh-qa/files/7259160/ResultsReportChanges-PreFix.zip) | 2021/09/29 | @Deblintrake09 | :red_circle: | without deactivating modules SCA, Rootcheck, Syscollector or Active Response.
| | 2021/09/29 | @Deblintrake09 | :green_circle:  | Deactivating SCA, Rootcheck, Syscollector or Active Response.

By deactivating the modules, tests pass. More results can be found in the related PR.

**Comment by damarisg** [comment]
I am attaching the result after it was merged, to verify that it continues to work.

| Results | Date | By | Status |
|:--:|:--:|:--:|:--:|
| [R-1886](https://github.com/wazuh/wazuh-qa/files/7281884/R-1886.zip) | 2021/10/04| Seyla | 🟢 | 


It closed by #1953
